packages/@lwc/engine-core/src/framework/api.ts

```diff
@@ -172,6 +181,14 @@ const ElementHook: Hooks<VElement> = {
         removeNodeHook(vnode, parentNode);
         removeElmHook(vnode);
     },
+    hydrate: (vnode, node) => {
+        vnode.elm = node as Element;
+
+        createElmHook(vnode);
+
+        // hydrate children hook
+        hydrateElementChildrenHook(vnode);
```

Comment:
The `hydrate` hook is not in charge of recursively invoking children `hydrate` if the diffing algo's job. 
This hook should only be in charge of upgrading a plain element node with props and event listeners that can't be serialized in HTML.